### PR TITLE
IsItUp: Add support for 'is it down' forms

### LINKED ISSUE #2837: IsItUp: Support 'is it down' forms
I can never remember if its 'up' or 'down', so being able to type `is it down ...` would be mighty handy.

See, for example: https://duckduckgo.com/?q=isitdown+duckduckgo.com&ia=web

---

IA Page:  http://duck.co/ia/view/is_it_up 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @rasikapohankar

### PR DESCRIPTION
**Pull request by rasikapohankar**
## Description of new Instant Answer, or changes

Adds support for triggers such as:
1. `isitdown`, 'is it down'
2. `isitup`, `is it up`
3. `is X up right now`
4. `status of X`

## Related Issues and Discussions

Fixes #2837 (finish #2955)

## People to notify

@GuiltyDolphin

---

Instant Answer Page: https://duck.co/ia/view/is_it_up

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by GuiltyDolphin** [comment]
@rasikapohankar Thank you!

File: lib/DDG/Spice/IsItUp.pm [hunk 1]
```diff
@@ -8,21 +8,32 @@ use Net::IDN::Encode qw(domain_to_ascii domain_to_unicode);
...
+
+    my ($domain, $ascii, $root_url);
     my $publicSuffix = Domain::PublicSuffix->new();
 
-    my $root_url = $_[1];
+    my $query = $_;
+
+    $query =~ $query_regex;
+
+    $root_url = $2;
```
Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: @rasikapohankar @GuiltyDolphin `$2` isn't guaranteed to be defined. We need to `return` unless the regex matches. Now that we're using trigger words, there are a lot more false positive queries hitting the regex.

I'm seeing a lot of `Use of uninitialized value $length in numeric gt (>) at /~~~~/Domain/PublicSuffix.pm line 330.` errors

Comment (rasikapohankar) [inline lib/DDG/Spice/IsItUp.pm]: @moollaza Thanks a lot for pointing it out. I'll make the change.

Can you please share which queries gave this error? I saw it on `is .com up`, did you see the error on a similar query?

**Comment by moollaza** [comment]
> Can you please share which queries gave this error? I saw it on is .com up, did you see the error on a similar query?

@rasikapohankar I tried random queries containing our trigger words but not containing a url:

```
Query: is testing this thing up
  You entered: is testing this thing up
Use of uninitialized value $length in numeric gt (>) at /~~~~/Domain/PublicSuffix.pm line 330.
```

**Pull request merged**